### ISSUE TITLE
Result error when chinese site name been set

### ISSUE THREAD

**Issue by OREOCODEDEV** [upptime/upptime#933]
**Describe the bug**
When chinese site name been set (i think other language will cause this bug too), upptime will only process ascii character in site name when generate graphs and history records.
This cause wrong graphs and result on status page, details below.

**To Reproduce**
Set chinese site name in .upptimerc.yml

**Screenshots**
Here is my status page generated by upptime [https://oreocodedev.github.io/upptime/](https://oreocodedev.github.io/upptime/)
![image](https://github.com/upptime/upptime/assets/44545625/5f891269-b883-485a-8da0-7db8b60263b0)
I set four different site, each of them have different site name, but in status page, two sites with different site names but containing the same ascii characters got same graphs and results, and the site which have no ascii character in site name, have no graphs shown.
I thought this will cause some result errors when more than one pure chinese site name where been set, is there have any ways to solve this problem?

**Comment by AnandChowdhary** [comment]
Aah you're totally right about this. We use @sindresorhus/slugify to slugify and I think it removes non-ASCII characters. What you can do is supply a custom slug to fix this:

```yml
sites:
  - name: Google
    slug: google-example
    url: https://www.google.com
```

In the above example, `slug` will be used that you provided rather than trying to generate it from the title.

**Comment by OREOCODEDEV** [comment]
Many thanks for your reply, got correct result by setting `slug` configuration.

I noticed that `Graphs CI` seems still using slugify string as graph file path after setting `slug` (in commit [0ff1496](https://github.com/OREOCODEDEV/upptime/commit/0ff14960587602986d0a1f7747e21c42e4d7cd38)), cause no graph shown in status page, and in action runs [Graphs CI](https://github.com/OREOCODEDEV/upptime/actions/runs/7982454170/job/21796002585), git rejected push command because some conflict content, did I perform certain step incorrectly?

**Comment by ttasovac** [comment]
I can also confirm that graphs don't get generated on the status page for site names written in Hebrew characters —  despite the use of an additional ascii slug field.

<img width="768" alt="Снимок экрана 2024-07-12 в 08 19 11" src="https://github.com/user-attachments/assets/1d5bc3ad-76ed-401d-abfb-be70d1f040dd">


@AnandChowdhary, would you consider releasing a new version of `upptime/graphs` to make available the fix introduced in https://github.com/upptime/graphs/commit/aac906c95084710c2e13b3d83a4499612d2a804d as reported by @asasenaru above?

**Comment by AnandChowdhary** [comment]
Yes, absolutely, this is important and needs to get released asap!

**Comment by AnandChowdhary** [comment]
I've been trying to get this out but with no success, the dependencies are old and `gyp` needs to build with new Node versions (how it works under the hood is using chart js canvas on Node to render images), so I'm thinking that maybe it's best to deprecate the entire upptime/graphs repo and use browser-based charts to render these rather than as images (also better for the git repo size since we don't generate and store PNGs).

**Comment by OREOCODEDEV** [comment]
> I've been trying to get this out but with no success, the dependencies are old and `gyp` needs to build with new Node versions (how it works under the hood is using chart js canvas on Node to render images), so I'm thinking that maybe it's best to deprecate the entire upptime/graphs repo and use browser-based charts to render these rather than as images (also better for the git repo size since we don't generate and store PNGs).

Glad to hear there's progress on the issue, sounds like there are going to be a lot of changes coming 🚀

**Comment by gongfuture** [comment]
Is there any updates?
